study-builder/studies/testboston/study-events.conf

```diff
@@ -238,6 +252,19 @@
       "dispatchToHousekeeping": false,
       "order": 1
     },
+    {
+      "trigger": {
+        "type": "ACTIVITY_STATUS",
+        "activityCode": ${id.act.baseline_symptom},
+        "statusType": "CREATED"
+      },
+      "action": {
+        "type": "HIDE_ACTIVITIES",
```

Comment:
Do you think there's an alternative approach here, of extending the `HIDE_ACTIVITIES` event instead of adding `hide_instances` flag? What if we add a new flag to `HIDE_ACTIVITIES` to "hide existing instances except the latest one". This way, we can chain this event after an `INSTANCE_CREATION` event to hide any old instances while keeping the latest instance (the one just created) visible.

The implementation will be totally different, and probably a bit more work. But the good thing is we can use this for both adhoc_symptom and longitudinal_covid activities. Two approaches I see:

1. Add a flag to the event action itself. We'll need to create a new table for `HIDE_ACTIVITIES` event action to hold this new boolean flag because we don't have a table yet. The drawback here is that this flag will apply for every activity listed in `activityCodes`.

2. We make it granular and specific to each activity we list. So we deprecate `activityCodes` and add `activities` instead, which will make it look like this:
    ```
    "action": {
      "type": "HIDE_ACTIVITIES",
      "activities": [
        { "activityCode": "foobar", "excludeLatest": true },
        ...
      ]
    }
    ```
    For this, we can add a column to the `event_action_target_activity` table. The downside here is that the `event_action_target_activity` table is also used in the `MARK_ACTIVITIES_READ_ONLY` event action. But if we named the column `exclude_latest_instance` then it should also make sense for `MARK_ACTIVITIES_READ_ONLY` too? (we don't have to support excluding latest for `MARK_ACTIVITIES_READ_ONLY` right now but can do so later).

Reply:
Approach I took seem to be more cleaner & less confusing to me than hacking existing events.